st3/lsp_utils/_client_handler/abstract_plugin.py

```diff
@@ -128,12 +128,19 @@ def can_start(cls, window: sublime.Window, initiating_view: sublime.View,
     @classmethod
     def on_pre_start(cls, window: sublime.Window, initiating_view: sublime.View,
                      workspace_folders: List[WorkspaceFolder], configuration: ClientConfig) -> Optional[str]:
-        extra_paths = path.pathsep.join(cls.get_additional_paths())
-        if extra_paths:
-            original_path = configuration.env.get('PATH') or ''
-            if isinstance(original_path, list):
-                original_path = path.pathsep.join(original_path)
-            configuration.env['PATH'] = path.pathsep.join([extra_paths, original_path])
+        original_path_raw = configuration.env.get('PATH') or ''
+        if isinstance(original_path_raw, str):
+            original_paths = original_path_raw.split(path.pathsep)
+        else:
+            original_paths = original_path_raw
+
+        # To fix https://github.com/TerminalFi/LSP-copilot/issues/163 ,
+        # We don't want to add the same path multiple times whenever a new server session is created.
+        # Note that additional paths should be prepended to the original paths.
+        wanted_paths = [path for path in cls.get_additional_paths() if path not in original_paths]
```

Comment:
All this logic is not really necessary when `cls.get_additional_paths()` is empty. So like before, I'd prefer to avoid it when not needed. Either by running this conditionally or having an early return.